### DISCUSSION TITLE
Change button icon based on process (swayidle) running or not

### DISCUSSION THREAD

**Discussion by mrbman** [elkowar/eww#767] [General]
I'm really struggling. What I am trying to do is create a button (icon) that when on-click runs a script, so far so simple. The script can be anything but in my scenario I want to toggle swayidle running or not (inhibit idle), and for the icon of the button to change (and change color) to reflect the state of the process (swayidle), either running or not.

This should be simple but I just don't seem to be able to get my head around it. Any help would be greatly appreciated!

**Comment by charlesrocket** [comment]
You could use ternary op to set labels depending on a specific state: https://github.com/charlesrocket/dotfiles/blob/7cdfdb9ef8c6aecb12e62cd7a75b1abcb91452b5/.config/eww/eww.yuck#L98-L102

Then update the state of the variable with `eww update buttonicon=green`.
